### DISCUSSION TITLE
Different Versions of Gen 2?

### DISCUSSION THREAD

**Discussion by copekyle** [emporia-vue-local/esphome#82] [General]
So I recently upgraded my install from one to two devices. I purchased the second device from an auction site and realized that it was built different from the device I purchased directly from Emporia. A few noticeable differences:

1. CT sensors had emporia logo printed on them and also didn't have a stamp for the direction to install towards the breaker. The new version has both K>L and Breaker> stamps, whereas this older unit just has the K>L stamp.
2. CT Sensors on older version have a thicker wire or wire jacket. I didn't cut them open to see
3. Power cable on the older version has a cable bundle wrap around it AND it has a disconnect in the middle of the power cable. Newer version just has wires and no disconnect
4. Antenna plug on the older version is designed to install the antenna from outside of enclosure into the inside. The plug on the newer one is opposite (install from inside of enclosure)
5. FINALLY and the MAIN REASON I'm posting all this. The red and black wires are swapped in the programming from old to new. In the OLD version, the Black wire is phase B and the Red wire is Phase A. The New version, these are swapped. In installed both of them based on the new version (because that's what I originally installed) and when I booted them up, the old version only took measurements on my balanced 240V circuits. I went into the code and swapped phase a and phase b and everything came online.

Moral of the story (and some of you might have already been through this), there are two versions of Gen 2 that are physically wired differently. Don't make the same mistake I did.

**Comment by flaviut** [comment]
Interesting! Did you notice any kind of revision information being printed on it?

**Reply by airdrummingfool** [reply]
I just flashed my first Emporia Vue 2, and it corresponds to the "new" version described in the OP (includes "Breaker>" stamps on CT clamps, antenna plug installs from inside, black wire is Phase A). The board has "REV 5" printed on it.

**Reply by airdrummingfool** [reply]
![IMG_9031](https://user-images.githubusercontent.com/631249/222940629-5a72bfb2-1d2a-403c-8fe2-fdfe59cf9693.jpeg)

**Comment by copekyle** [comment]
No difference in the packaging/case/labels. I had already got them working by the time I posted this and so didn't want to open them back up to look at the circuit boards.

**Reply by flaviut** [reply]
Makes sense! Thanks for checking, hopefully someone else with the new version will get some board pictures before installing.

**Comment by ChrisR000** [comment]
I’m pretty sure I have the updated version and I can’t get any current/power measurements.  Is there something I need to adjust? Voltage reads fine. 

*EDIT* switched the red and black in the yaml and got the readings.

**Comment by jklap** [comment]
Hmmmm... so this is also likely the reason why you can flip the CT clamps and you start to see power on Phase A and B....

SOOOO... what would the impact back of leaving the clamps flipped and the programming the same? Because I just put everything back together right before I read this :(

**Comment by jcmilley** [comment]
I read this with interest because I just stumbled on the red/black flip between my older rev 4 and the new rev 5.  Given I just opened up the rev 4 today to 'esphome it' I figured I would share the pics of the device just to have both revisions in this thread.

I'll will add that my stare and compare of the two revisions did not highlight anything notable.  The soldering on my rev 4 looks to be a little sloppy but the traces for the power connector at the bottom look to be the same as the rev 5.

![vue-rev4-top](https://github.com/user-attachments/assets/e43db5cd-3a79-4481-8e9c-13013e90a8b0)
![vue-rev4-bottom](https://github.com/user-attachments/assets/5b3c6753-3309-4351-b87c-684145aa4c5f)
